src/chainstate/stacks/bench.rs

```diff
@@ -0,0 +1,882 @@
+#[cfg(test)]
+mod tests {
+    use crate::{
+        chainstate::{
+            burn::db::sortdb::SortitionDB,
+            stacks::{
+                db::{blocks::MemPoolRejection, StacksChainState},
+                miner::{
+                    test::{
+                        get_stacks_account, make_coinbase, make_contract_call, make_token_transfer,
+                        make_user_contract_call, make_user_contract_publish,
+                        make_user_stacks_transfer,
+                    },
+                    BlockBuilderSettings,
+                },
+                StacksBlock, StacksBlockBuilder, StacksBlockHeader,
+            },
+        },
+        core::{mempool::MemPoolWalkSettings, MemPoolDB},
+        net::test::{TestPeer, TestPeerConfig},
+    };
+
+    use super::*;
+    use clarity::{
+        address::{AddressHashMode, C32_ADDRESS_VERSION_TESTNET_SINGLESIG},
+        types::{
+            chainstate::{ConsensusHash, StacksAddress, StacksPrivateKey, StacksPublicKey},
+            Address, StacksEpochId,
+        },
+        vm::{costs::ExecutionCost, types::StacksAddressExtensions, Value},
+    };
+    use rand::Rng;
+    use stacks_common::util::hash::Hash160;
+
+    #[test]
+    #[ignore]
+    /// Give the miner an unlimited time to build a full block, then check how long it took
+    /// and how many transactions were included.
+    /// The transactions are a random mix of NFT mints, NFT transfers, and STX transfers.
+    /// The information we are looking for can be found in the logs, for example:
+    ///   Miner: mined anchored block, block_hash: 5c259637d326c86bac17900d07f4d0d38b8cf5b854b6898bdf51017c2cab6461,
+    ///     height: 2,   tx_count: 10567, parent_stacks_block_hash: 34553785fcaa07bb07cf3341e30c18ba390381591f10e502f27c346c50d00cf7,
+    ///     parent_stacks_microblock: 0000000000000000000000000000000000000000000000000000000000000000,
+    ///     parent_stacks_microblock_seq:    0, block_size: 2097103,
+    ///     execution_consumed: {"runtime": 22269784000, "write_len": 82452, "write_cnt": 12916, "read_len": 17205312, "read_cnt": 42972},
+    ///     assembly_time_ms: 99140, tx_fees_microstacks: 18138000
+    /// where the interesting fields are "assembly_time_ms" and "tx_count".
+    ///
+    /// To run this test:
+    ///   cargo test --package stacks-subnets --lib -- chainstate::stacks::bench::tests::test_max_block --exact --nocapture --ignored
```

Comment:
Doc comments are rendered as Markdown by `cargo doc`, so use appropriate syntax. Same for other comments:

```suggestion
    /// ```
    ///   Miner: mined anchored block, block_hash: 5c259637d326c86bac17900d07f4d0d38b8cf5b854b6898bdf51017c2cab6461,
    ///     height: 2,   tx_count: 10567, parent_stacks_block_hash: 34553785fcaa07bb07cf3341e30c18ba390381591f10e502f27c346c50d00cf7,
    ///     parent_stacks_microblock: 0000000000000000000000000000000000000000000000000000000000000000,
    ///     parent_stacks_microblock_seq:    0, block_size: 2097103,
    ///     execution_consumed: {"runtime": 22269784000, "write_len": 82452, "write_cnt": 12916, "read_len": 17205312, "read_cnt": 42972},
    ///     assembly_time_ms: 99140, tx_fees_microstacks: 18138000
    /// ```
    /// where the interesting fields are `assembly_time_ms` and `tx_count`.
    ///
    /// To run this test:
    /// ```sh
    /// cargo test --package stacks-subnets --lib -- chainstate::stacks::bench::tests::test_max_block --exact --nocapture --ignored
    /// ```
```

Reply:
Good point. Added.